File: src/BetterLyrics.Sdk/Interfaces/Plugins/IConfigurator.cs
using BetterLyrics.Sdk.Enums;

namespace BetterLyrics.Sdk.Interfaces.Plugins;

public interface IConfigurator
{
    object Get(string key, object defaultValue);
    void Set(string key, object value, ConfigChangedBy configChangedBy);

    event EventHandler<string, ConfigChangedBy>? OnConfigChanged;
}

File: src/BetterLyrics.KMP/shared/src/commonMain/kotlin/com/jayfunc/betterlyrics/sdk/interfaces/plugins/IConfigurator.kt
package com.jayfunc.betterlyrics.sdk.interfaces.plugins

import com.jayfunc.betterlyrics.sdk.enums.ConfigChangedBy

interface IConfigurator {
    fun get(key: String, defaultValue: Any?): Any?

    fun set(key: String, value: Any?, configChangedBy: ConfigChangedBy)

    // Translating C# 'event' to standard Kotlin listener registration
    fun addOnConfigChangedListener(listener: (String, ConfigChangedBy) -> Unit)

    fun removeOnConfigChangedListener(listener: (String, ConfigChangedBy) -> Unit)
}
